tests/RateLimiter/CounterTest.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Yiisoft\Yii\Web\Tests\RateLimiter;
+
+use InvalidArgumentException;
+use PHPUnit\Framework\TestCase;
+use Yiisoft\Cache\ArrayCache;
+use Yiisoft\Yii\Web\RateLimiter\Counter;
+
+final class CounterTest extends TestCase
+{
+    /**
+     * @test
+     */
+    public function statisticsShouldBeCorrectWhenLimitIsNotReached(): void
+    {
+        $counter = new Counter(2, 5, new ArrayCache());
+        $counter->setId('key');
+
+        $statistics = $counter->incrementAndGetResult();
+        $this->assertEquals(2, $statistics->getLimit());
+        $this->assertEquals(1, $statistics->getRemaining());
+        $this->assertEquals(2500, $statistics->getReset());
+        $this->assertFalse($statistics->isLimitReached());
```

Comment:
add more loop
```php
$statistics = $counter->incrementAndGetResult();
$this->assertEquals(2, $statistics->getLimit());
$this->assertEquals(1, $statistics->getRemaining());   // <--- remaining = 1
$this->assertEquals(2500, $statistics->getReset());
$this->assertFalse($statistics->isLimitReached());
// next loop
$statistics = $counter->incrementAndGetResult();
// ...
```

Reply:
0 is when limit is reached. It's out of the scope of this test.